ksqldb-functional-tests/src/test/resources/query-validation-tests/none.json

```diff
@@ -0,0 +1,181 @@
+{
+  "comments": [
+    "NONE format testing."
+  ],
+  "tests": [
+    {
+      "name": "as key format of keyless stream",
+      "statements": [
+        "CREATE STREAM INPUT (foo INT) WITH (kafka_topic='input_topic', key_format='NONE', value_format='JSON');",
+        "CREATE STREAM OUTPUT AS SELECT * FROM INPUT;"
+      ],
+      "inputs": [
+        {"topic": "input_topic", "key": null, "value": {"foo": 10}},
+        {"topic": "input_topic", "key": "data that should be ignored", "value": {"foo": 11}},
+        {"topic": "input_topic", "key": null, "value": null}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": null, "value": {"FOO": 10}},
+        {"topic": "OUTPUT", "key": null, "value": {"FOO": 11}},
+        {"topic": "OUTPUT", "key": null, "value": null}
+      ]
+    },
+    {
+      "name": "as key format of keyed stream",
+      "statements": [
+        "CREATE STREAM INPUT (K STRING KEY, foo INT) WITH (kafka_topic='input_topic', key_format='NONE', value_format='JSON');"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "The 'NONE' format can only be used when no columns are defined. Got: [`K` STRING KEY]"
+      }
+    },
+    {
+      "name": "as key format of table",
+      "statements": [
+        "CREATE TABLE INPUT (K STRING PRIMARY KEY, foo INT) WITH (kafka_topic='input_topic', key_format='NONE', value_format='JSON');"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "The 'NONE' format can only be used when no columns are defined. Got: [`K` STRING KEY]"
+      }
+    },
+    {
+      "name": "as value format with value columns",
```

Comment:
Can we add a test for what happens if someone tries to set NONE as value format with no value columns? I assume this is not supported. It'd be good to check that the error message is sensible.

Reply:
You'd think no value columns would not work right?  Nope... it works, and just fails if you try and use the source.

(See https://github.com/confluentinc/ksql/issues/6352)

Reply:
That's unfortunate... is it worth slipping a line into the docs for the new `NONE` format to clarify that ksqlDB doesn't yet support empty value columns, so the `NONE` format should only be used as key format for now? Could also fail up front if users try to use the `NONE` format as a value format but that might not be worth the effort since it's code that will eventually be removed.